.pre-commit-config.yaml

```diff
@@ -59,3 +59,18 @@ repos:
           - -w      # autofix in place
           - -d      # show diff
           - -i=2    # indent 2 spaces
+
+  # Example usage of dotnet-format: pre-commit run dotnet-format --all-files
+  - repo: local
+    hooks:
+      - id: dotnet-format
+        name: dotnet-format
+        entry: script/check-dotnet-format.sh
+        language: script
+        types:
+          - file
+          - text
+        pass_filenames: false
```

Comment:
Does `dotnet format` have a mode where it can receive multiple filenames to check? That would be more efficient if so.

Reply:
`dotnet format` requires the project file (or all files to be within the same folder). For example:
```
PS C:\Users\Liz\Documents\GitHub\pre-commit-plugins> dotnet-format --check --include .\samples\Testing.cs .\samples\TestAgain.cs
  Could not find a MSBuild project file or solution file in 'C:\Users\Liz\Documents\GitHub\pre-commit-plugins'. Specify which to use with the --workspace option.
```
If the `--workspace` option is set, `dotnet format` will only run over one workspace. I tried matching `.csproj` files and running `dotnet format` over each directory above that file but was unable to resolve all bugs. If this would be preferred, I'm happy to close the PR, fix that version and reopen it.

Reply:
^ For ease of future use, I implemented the above. If no arguments are provided, the default behaviour is to run over any C# files detected. Otherwise, by adding the `.csproj` file for each C# project to be linted as an argument, `dotnet format` is restricted to those files listed.